crypto/txscript/src/opcodes/mod.rs

```diff
@@ -210,10 +212,33 @@ fn push_number<T: VerifiableTransaction, Reused: SigHashReusedValues>(
     number: i64,
     vm: &mut TxScriptEngine<T, Reused>,
 ) -> OpCodeResult {
-    vm.dstack.push_item(number);
+    if number == i64::MIN {
```

Comment:
Isn't this check redundant given the new check within `push_item`?

Reply:
nah. push_item serializes first, then can produce error. this check can prewvent serialization. so its an optimization